java-checks/src/main/java/org/sonar/java/checks/PrintStackTraceCalledWithoutArgumentCheck.java

```diff
@@ -77,19 +71,7 @@ private boolean enclosingClassExtendsThrowable() {
     return enclosingClass.peek() != null && enclosingClass.peek().type().isSubtypeOf("java.lang.Throwable");
   }
 
-  private static boolean useDefaultOutputStream(Arguments arguments) {
-    if (!arguments.isEmpty()) {
-      ExpressionTree firstArg = arguments.get(0);
-      if (firstArg.is(Tree.Kind.MEMBER_SELECT)) {
-        MemberSelectExpressionTree mset = (MemberSelectExpressionTree) firstArg;
-        ExpressionTree expression = mset.expression();
-        String identifier = mset.identifier().name();
-        return expression.is(Tree.Kind.IDENTIFIER)
-          && "System".equals(((IdentifierTree) expression).name())
-          && ("err".equals(identifier) || "out".equals(identifier));
-      }
-      return false;
-    }
-    return true;
+  private static boolean calledOnTypeInheritedFromThrowable(MethodInvocationTree tree) {
```

Comment:
I would pass directly the methodSelect to this method, that should be clearer than the safe implicit cast.